Assets/Scripts/NPCs/StateMachine/States/Wander.cs

```diff
@@ -113,6 +113,11 @@ private void MoveTowardsTarget()
 
         private bool TargetReached()
         {
+            if (_pathToTarget == null)
```

Comment:
I think that this is just putting a band aid on the problem (which doesn't reaallyyy help because the list is initialized so you should also check Count < 1). The problem is that the NPCs are not being removed and reinitialized once the player decides to play again.

Reply:
Changing the script order in the meta files seems to resolve this issue entirely.